app/src/main/java/com/android/unio/ui/association/AssociationProfile.kt

```diff
@@ -412,14 +598,402 @@ private fun AssociationMembers(
                     Text("$firstName $lastName")
 
                     // Role Badge
-                    RoleBadge(member.role)
+                    val association = associationViewModel.selectedAssociation.value
+                    val userRole =
+                        association?.roles?.find {
+                          it.uid == association.members.find { it.uid == user.value?.uid }?.roleUid
+                        }
+
+                    if (userRole != null) {
+                      RoleBadge(userRole)
+                    }
                   }
                 }
               }
         }
       }
 }
 
+/**
+ * Displays a list of members of an association and allows searching for members using a search bar.
+ * When a member is selected from the search, the view scrolls to the respective member's card.
+ * Additionally, it allows managing the association's roles.
+ *
+ * @param associationViewModel The ViewModel responsible for managing the association's data.
+ * @param userUid The unique identifier of the current user.
+ * @param onMemberClick A lambda function to be called when a member's card is clicked, passing the
+ *   selected member's data.
+ * @param searchViewModel The ViewModel responsible for managing member search functionality.
+ */
+@Composable
+private fun AssociationActionsMembers(
+    associationViewModel: AssociationViewModel,
+    userUid: String,
+    onMemberClick: (User) -> Unit,
+    searchViewModel: SearchViewModel,
+) {
+  val context = LocalContext.current
+
+  val association by associationViewModel.selectedAssociation.collectAsState()
+  val members = association?.members
+  val pagerState = rememberPagerState(initialPage = 0) { members?.size ?: 0 }
+  val coroutineScope = rememberCoroutineScope()
+
+  val searchBar: @Composable () -> Unit = {
+    association?.let {
+      MemberSearchBar(
+          searchViewModel = searchViewModel,
+          associationUid = it.uid,
+          userUid = userUid,
+          onMemberSelected = { selectedMember ->
+            val targetPage = members?.indexOfFirst { it.uid == selectedMember.uid }
+            if (targetPage != null && targetPage >= 0) {
+              coroutineScope.launch { pagerState.animateScrollToPage(targetPage) }
+            }
+          },
+          shouldCloseExpandable = false,
+          onOutsideClickHandled = {})
+    }
+  }
+
+  val cardContent: @Composable (Member) -> Unit = { member ->
+    val user = associationViewModel.getUserFromMember(member).collectAsState()
```

Comment:
Again, consider using `by`:
```kotlin
val user by associationViewModel.getUserFromMember(member).collectAsState()
```
And remove `.value` calls.

Also, consider returning early if `user == null`, so that you can use `!!` assertions instead of nested `.let` calls.

Reply:
Yep ! I think you resolved a big problem I had struggled to find out, because after too much time code I can't make the difference between `val user = associationViewModel.getUserFromMember(member).collectAsState()` and `val user by associationViewModel.getUserFromMember(member).collectAsState()` haha